TESTDATA_GUIDELINES.md

```diff
@@ -0,0 +1,277 @@
+device (enable USB debugging!) and install the application. It's a good idea to go into the app settings and grant all
+possible permissions to the app, to prevent popups from getting in the way of your Puma script.
+
+If this is the first time you're running your script, we recommend 'babysitting' it: it is very possible that some
+actions will fail, and when you see what happens it'll be easier to fix your code. Sometimes you can also manually
+intervene and do a tap that you forgot to put in your Puma code.
+
+If you are repeating this experiment, after babysitting your puma script once or twice all the problems should be ironed
+out and a complete hands-off approach should be possible: just start the script and go get some coffee!
+
+### Creating test data manually
+
+We highly recommend to manually create test data with two people: an operator who operates the device(s), and a
+coordinator who handles the script. The coordinator tells the operator which action to take next, and annotates the
+script with the time stamps. In case the operator makes a mistake (makes a typo or executes the wrong user action in
+the UI, or perhaps the coordinator gave th wrong instruction), the coordinator also writes this down.
+
+At first glance it might seem overkill to have two people 'just play around with a phone', but a mistake is made **very
+easily**. Having two people in this process greatly reduces the number of mistakes you will make (yes, mistakes will be
+made).
+
+By having the coordinator annotate everything, you end up with a complete and correct script of all actions that were
+taken, including the time stamps. This is valuable metadata that you can store along your reference data after
+extraction, so when you analyze the reference data, you know exactly how it was created and what you could find in that
+data.
+
+## 3. Extracting and storing the test data
+
+### Extraction
+
+After you have executed the script, your reference data is now ready, you just need to extract it. How to do this
+depends on what data you need:
+
+1. If you do network analysis, I hope you didn't forget to record the network traffic!
+2. If you want a full device image, use your favorite tool to make this image
+3. If you only want specific app data, you'll need to delve into the device to find its location
+
+The first two cases are trivial, in the sense that if you're doing this sort of analysis you probably don't need any
+help to collect the data you want.
+
+In the last case, you might not know where to get the data. And you;re not alone: I don't know either. Where app store
+their data is dependent on the OS and the app itself.
+
+Luckily on modern mobile devices, apps cannot just store data anywhere they want. On android, relevant app data is
+usually stored in `/data/data/<package name of your app>`. Within this folder, the app can chose how to store its data.
+What we usually do is copy this entire folder and then explore this data.
```

Comment:
I think for iOS, we can add a similar path, right? We know where app data is generally stored.